src/services/importService.ts

```diff
@@ -0,0 +1,191 @@
+            exportFormat: null,
+            videoSettings: {
+                frameExtractionRate: 15,
+            },
+            autoSave: true,
+            assets: {},
+        };
+
+        const assetService = new AssetService(convertedProject);
+        generatedAssetMetadata = this.generateAssets(project, assetService);
+
+        const saveAssets = generatedAssetMetadata.map((assetMetadata) => {
+            assetMetadata.then((metadata) => {
+                return assetService.save(metadata);
+            });
+        });
+
+        try {
+            Promise.all(saveAssets);
+        } catch (e) {
+            throw e;
+        }
+
+        return convertedProject;
+    }
+
+    /**
+     * Generate connections from v1 project file location
+     * @param project - V1 Project Content and File Information
+     */
+    private generateConnections(project: any): IConnection[] {
+        const sourceConnection: IConnection = {
+            id: shortid.generate(),
+            name: `${project.file.name.split(".")[0]} Source Connection`,
+            providerType: "localFileSystemProxy",
+            providerOptions: {
+                folderPath: project.file.path.replace(/[^\/]*$/, ""),
+            },
+        };
+
+        const targetConnection: IConnection = {
+            id: shortid.generate(),
+            name: `${project.file.name.split(".")[0]} Target Connection`,
+            providerType: "localFileSystemProxy",
+            providerOptions: {
+                folderPath: project.file.path.replace(/[^\/]*$/, ""),
+            },
+        };
+
+        const connections: IConnection[] = [sourceConnection, targetConnection];
+
+        return(connections);
+    }
+
+    /**
+     * Parse v1 project's tag string and return array of ITags
+     * @param project - V1 Project Content and File Information
+     */
+    private parseTags(project: any): ITag[] {
+        const finalTags: ITag[] = [];
+        const tagStrings = project.inputTags.split(",");
+        const tagColors = project.tag_colors;
+
+        for (let i = 0; i < tagColors.length; i++) {
+            const newTag = {
+                name: tagStrings[i],
+                color: tagColors[i],
+            };
+            finalTags.push(newTag);
+        }
+        return finalTags;
+    }
+
+    /**
+     * Generate assets based on V1 Project frames and regions
+     * @param project - V1 Project Content and File Information
+     */
+    private generateAssets(project: any, assetService: AssetService): Array<Promise<IAssetMetadata>> {
+        let originalProject: IV1Project;
+        const generatedAssetMetadata: Array<Promise<IAssetMetadata>> = [];
+        let generatedRegion: IRegion;
+        let assetState: AssetState;
+        const currentTagColorIndex = randomIntInRange(0, TagColors.length);
+
+        originalProject = JSON.parse(project.content);
+
+        for (const frameName in originalProject.frames) {
```

Comment:
This loop is rather complex with lots of inner loops going on... Let's refactor to split some of this log out into smaller bitsize methods that you can then reuse here.

Reply:
I would break this down into many smaller methods that convert v1Type to v2Type.  Then this function should essentially becoming a simple orchestrator to convert the project.